### ISSUE TITLE
hasMany async collection items render twice when using `store.push`

### ISSUE THREAD

**Issue by SeyZ** [emberjs/data#2757]
Hello,

Here's the demonstration of the bug: http://jsbin.com/doraha/5/edit?html,js,output

I would expect to see something like:

```
BAR 1
BAR 2
BAR 3
BAR 4
...
```

instead of

```
BAR 1
BAR 2
BAR 3
BAR 4
...
BAR 4
BAR 3
BAR 2
```

It works well if I set the `hasMany` to `async: false` (by default).

**Comment by SeyZ** [comment]
FYI: It works well with `async: true` if I switch the ember-data to `1.0.0-beta.12`. The bug appears from `.0.0-beta.14`

**Comment by jbescoyez** [comment]
@bmac This bug looks like https://github.com/emberjs/data/issues/2740 and https://github.com/emberjs/data/issues/2713 . Still have to check if https://github.com/emberjs/data/pull/2723 fixes it.

**Comment by bmac** [comment]
I just tested this jsbin agains #2723 and that pr should fix this issue.

**Comment by SeyZ** [comment]
Awesome @bmac !

**Comment by bmac** [comment]
Fixed by #2723

**Comment by jbescoyez** [comment]
Many tnx @bmac
